### PR TITLE
Add front-end tests

### PR DESCRIPTION
**Pull request by txels**
I've been willing to get some front-end testing in for a while. This adds the first JS unit tests and the `karma` test runner is executed as part of the tox build. As a bonus I've also added `jshint` to the build.

### CODE REVIEW

**Comment by alexcouper** [comment]
When I run the tests using `tox` I get 3 errors and 2 failures in the python tests.

Am I doing something wrong?

**Comment by alexcouper** [comment]
``` diff
diff --git a/tox.ini b/tox.ini
index 308f304..493c6b1 100644
--- a/tox.ini
+++ b/tox.ini
@@ -2,13 +2,17 @@
 envlist = py27

 [testenv]
-deps =
+setenv =
+    CFLAGS=-Qunused-arguments
+    CPPFLAGS=-Qunused-arguments
+deps =
     -r{toxinidir}/requirements/dev.txt
-commands =
+commands =
     flake8 ployst
     jshint ployst
     karma start --single-run
-    nosetests -s  # --with-coverage --cover-package=ployst --cover-html
+    python manage.py test
     sphinx-build -b html docs/source docs/build

 [flake8]
```

This is the output I get from `tox` after I make the above changes:

screen shot 2014-04-08 at 10 16 03 pm

Any idea how to remove the jshint/karma warnings?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+language: python
+python: 2.7
+install:
+  - pip install tox
+  - npm install -d
```
Comment (txels) [inline .travis.yml:5]: I haven't found what this option means... is this "devDependencies"?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+language: python
+python: 2.7
+install:
+  - pip install tox
```
Comment (txels) [inline .travis.yml]: should we specify a version? (e.g. I had 1.4 and the whitelist didn't work, had to upgrade to 1.7.x)

**Comment by txels** [comment]
Nice to see Travis reporting green builds, cheers!

File: .travis.yml (hunk 1 above)
Comment (alexcouper) [inline .travis.yml:5]: It was used in an example I was looking for for package.json installs. Turns out it's just setting  log level to info.

https://www.npmjs.org/doc/misc/npm-config.html#Shorthands-and-Other-CLI-Niceties

File: .travis.yml (hunk 2 above)
Comment (alexcouper) [inline .travis.yml]: Probably worthwhile in that case!

Comment (txels) [inline .travis.yml]: done.

**Comment by txels** [comment]
I think this can be merged.

**Comment by alexcouper** [comment]
I'll merge once travis confirms life is good

**Pull request merged**